ci/images/ci-runner/hack/bin/deploy-cluster.sh

```diff
@@ -14,58 +14,93 @@ source "$SCRIPT_DIR/utils.sh"
+}
 
-    printf "HostedCluster CR Status: " | indent 4
-    kubectl -n clusters get hostedcluster "$CLUSTER_NAME" -o yaml
-
-    printf "Control Plane pods of HostedCluster: " | indent 4
-    kubectl -n "clusters-${CLUSTER_NAME}" get pods
+# Add a function to check all of clusteroperators till AVAILABLE are true
+check_clusteroperators() {
+    operators=$(kubectl get co -o jsonpath='{range .items[*]}{.metadata.name}{" "}{end}')
 
-    # Export kubeconfig of hosted cluster
-    hypershift create kubeconfig --name "$CLUSTER_NAME" > "${CLUSTER_NAME}.kubeconfig"
+    for operator in $operators; do
+        # need to check if the operator is available or not in a loop
+        retries=0
+        max_retries=10
+        # if the operator is not available, wait 60s and check again
+        while [ "$retries" -lt "$max_retries" ]; do
+            available=$(kubectl get co "$operator" -o jsonpath='{.status.conditions[?(@.type=="Available")].status}')
+            if [ "$available" == "True" ]; then
+                break
+            fi
+            sleep 60
+            retries=$((retries + 1))
+            echo "Retried $retries times..."
+        done
+        # if the operator is still not available after 10 times, exit 1
+        if [ "$retries" -eq "$max_retries" ]; then
+            echo "Operator $operator is not available" >&2
+            exit 1
+        fi
+    done
 
-    printf "Cluster Operators on Hosted Cluster: " | indent 4
-    kubectl --kubeconfig="${CLUSTER_NAME}.kubeconfig" get co -o yaml
+    echo "All operators are available"
 }
 
 deploy_cluster() {
+    printf "Log in to your Red Hat account...\n" | indent 2
     setx_off
-    hypershift create cluster aws --pull-secret "$PULL_SECRET" --aws-creds "$AWS_CREDENTIALS"  --name "$CLUSTER_NAME"  --node-pool-replicas=2  --base-domain "$BASE_DOMAIN"  --region="$REGION" --release-image="$IMAGE" --root-volume-type=gp3 --root-volume-size=120 --instance-type=m5.2xlarge
+    rosa login --token="$ROSA_TOKEN"
     setx_on
 
-    echo "Wait until hypershift hosted cluster is ready..."
-    wait_period=0
-    while
-        [ \
-            "$(
-                kubectl -n clusters get hostedcluster "$CLUSTER_NAME" -o json \
-                    | jq -r '.status.version.history[0].state'
-            )" != "Completed" \
-        ]; do
-        if [ "$wait_period" -gt 1800 ]; then
-            echo "[ERROR] Failed to create OCP cluster." >&2
+    printf "Provision hypershift cluster...\n" | indent 2
+    rosa create cluster --cluster-name "$CLUSTER_NAME" \
+        --sts --mode=auto --oidc-config-id "24132snjs8gktc5rtufqv8hggjv3ivbt" \
+        --operator-roles-prefix plnsvc-ci --region "$REGION" --version "$OCP_VERSION" \
+        --compute-machine-type m5.2xlarge \
+        --subnet-ids="subnet-001487732ebdd14f4,subnet-0718fb663f4b97f38,subnet-0fe426997da62662c" \
+        --hosted-cp -y
+
+    printf "Wait till hypershift cluster is ready...\n" | indent 2
```

Comment:
till -> until?

Reply:
Fixed. I changed it to "Track the progress of the cluster creation...\n".